src/common/components/context-menu.js

```diff
@@ -134,6 +136,37 @@ function ContextMenu({ params, onClose }) {
 		}
 	}
 
+	// Select a menuitem from typing, similar to native context menus
+	function handleKeyDown(event) {
+		let { key } = event;
+		// Ignore non-characters
+		if (key.length !== 1 || !key.match(/\S/)) return;
+
+		// Clear search string after 3 seconds of no typing
+		if (searchTimeoutRef.current) {
+			clearTimeout(searchTimeoutRef.current);
+		}
+		searchTimeoutRef.current = setTimeout(() => {
+			searchStringRef.current = '';
+		}, 2000);
+
+		// Keep track of what has been typed so far
+		searchStringRef.current += key.toLowerCase();
+
+		// Find all buttons with text that start with what has been typed
+		let menuOptions = [...document.querySelectorAll(".context-menu button:not([disabled])")];
+		let candidates = menuOptions.filter(option => option.textContent.toLowerCase().startsWith(searchStringRef.current));
+		
+		// If there is only one candidate, click it right away
+		if (candidates.length == 1) {
+			candidates[0].click();
+		}
```

Comment:
But where is this concept from? It feels pretty unexpected that pressing a letter immediately activates an action.